pkg/blockbuilder/blockbuilder.go

```diff
@@ -169,47 +176,49 @@ func (b *BlockBuilder) starting(ctx context.Context) (err error) {
 	return nil
 }
 
-const (
-	// kafkaOffsetStart is a special offset value that means the beginning of the partition.
-	kafkaOffsetStart = int64(-2)
-
-	// kafkaOffsetEnd is a special offset value that means the end of the partition.
-	//kafkaOffsetEnd = int64(-1)
-)
-
-func (b *BlockBuilder) findOffsetsToStartAt(ctx context.Context, metrics *kprom.Metrics) (map[int32]kgo.Offset, error) {
+func (b *BlockBuilder) findOffsetsToStartAt(ctx context.Context, metrics *kprom.Metrics) (map[int32]kgo.Offset, int64, error) {
 	// We use an ephemeral client to fetch the offset and then create a new client with this offset.
 	// The reason for this is that changing the offset of an existing client requires to have used this client for fetching at least once.
 	// We don't want to do noop fetches just to warm up the client, so we create a new client instead.
 	cl, err := kgo.NewClient(commonKafkaClientOptions(b.cfg.Kafka, b.logger, metrics)...)
 	if err != nil {
-		return nil, fmt.Errorf("unable to create bootstrap client: %w", err)
+		return nil, -1, fmt.Errorf("unable to create bootstrap client: %w", err)
 	}
 	defer cl.Close()
 
 	admClient := kadm.NewClient(cl)
 
+	fallbackOffset := int64(b.cfg.LookbackOnNoCommit)
+	defaultKOffset := kgo.NewOffset()
+	if fallbackOffset < 0 {
+		defaultKOffset = defaultKOffset.At(fallbackOffset)
```

Comment:
Are we using `LookbackOnNoCommit` as a hack to set -1 and -2 to consume from beginning or end? Needs to be documented here in code and for the config.

Reply:
Yes, moving forward, I want to allow `b.cfg.LookbackOnNoCommit` to accept a special string value "start", that converts into `-1` (this also how kgo works with the offsets internally)